### PR TITLE
Add support for preloading v3 target state format

### PR DESCRIPTION
**Pull request by pipex**
Supervisor v13 and above uses the newer v3 target state endpoint. This updates the preloading mechanism so `apps.json` is created using this new format for OS images including this newer supervisor.

Change-type: minor

### CODE REVIEW

File: lib/preload.ts [hunk 1]
```diff
@@ -587,16 +609,32 @@ export class Preloader extends EventEmitter {
 					ci.image[0].is_stored_at__image_location.toLowerCase(),
 			});
 		});
+
 		// App from the state endpoint (v1 or v2 depending on the supervisor version).
-		const app = _.values(this.state.local.apps)[0];
 		if (this._supervisorLT7()) {
```
Comment (klutchell) [inline lib/preload.ts:614]:
```suggestion
		if (version === 1) {
```

File: lib/preload.ts [hunk 2]
```diff
@@ -587,16 +609,32 @@ export class Preloader extends EventEmitter {
...
+			const app = _.values(this.state.local.apps)[0];
 			// Pre-multicontainer: there is only one image: use the only image from the state endpoint.
 			images[0].is_stored_at__image_location = app.image.toLowerCase();
 		} else {
+			const _getAppServices = (apps) => {
+				if (this._supervisorLT13()) {
```
Comment (klutchell) [inline lib/preload.ts:620]:
```suggestion
				if (this._getStateVersion() === 2) {
```

File: lib/preload.ts [hunk 3]
```diff
@@ -587,16 +609,32 @@ export class Preloader extends EventEmitter {
 					ci.image[0].is_stored_at__image_location.toLowerCase(),
 			});
 		});
+
 		// App from the state endpoint (v1 or v2 depending on the supervisor version).
-		const app = _.values(this.state.local.apps)[0];
 		if (this._supervisorLT7()) {
+			const app = _.values(this.state.local.apps)[0];
```
Comment (klutchell) [inline lib/preload.ts:615]: Should we adjust `_getAppServices` to support all versions and use it everywhere for consistency?

Comment (pipex) [reply lib/preload.ts:615]: Yeah, that would simplify the code. I'll make the change

**Comment by klutchell** [comment]
LGTM. Have you been able to test it locally with a build of the CLI?

You can manually copy these files to the `node_modules` of a balena-cli clone and build it for testing. Or you can open a draft PR of the CLI and pin to this pre-release of the balena-preload package (`balena-preload@12.1.0-preload-v3-02d2264acd2e5ca01bb907d24426c4a3c919839d`).

**Comment by pipex** [comment]
@klutchell I have tested it (using [npm-link](https://docs.npmjs.com/cli/v8/commands/npm-link)). I tested with a v2 supervisor and a v3 supervisor. Both preload apps.json correctly, although the latest supervisor had a couple of preload bugs, hence https://github.com/balena-os/balena-supervisor/pull/1944

**Review by klutchell — APPROVED** [review]

**Pull request merged**